### PR TITLE
Add Events route

### PR DESCRIPTION
**Pull request by appleJax**
Added react templates for /events and /events/:id

Very basic layout, I will keep working on it. Just wanted to push up what I had because I rearranged some scaffolding that overlaps with what @jeremyjung is working on.

### CODE REVIEW

File: app/javascript/components/event.jsx [hunk 1]
```diff
@@ -2,46 +2,40 @@ import React from 'react'
...
+            <th />
+            {users.sort((a, b) => b.id - a.id)
+                  .map(user =>
+                    <th key={user.id} className='user'>{user.name}</th>
+            )}
+          </tr>
+        </thead>
+        <tbody>
+          {timeslots.map(timeslot =>
+            <Row key={timeslot.id} timeslot={timeslot} users={users} />
```
Comment (appleJax) [inline app/javascript/components/event.jsx]: I think we should sort the timeslots in increasing order. Not sure if the `start_time` field can be sorted like this, but something like:

```
 {timeslots.sort( (a, b) => a.start_time - b.start_time)
           .map( timeslot =>
             <Row key={timeslot.id} timeslot={timeslot} users={users} /> )
 }
```

**Comment by appleJax** [comment]
@jeremyjung great stuff so far! Lookin niiiice. 💯

**Comment by appleJax** [comment]
Ok, I think we've done a good bit of work for one PR. There's more work to be done, obviously, but I think this one is ready for review.

**Comment by gabrielbonner** [comment]
@appleJax Do you want to deal with the `Timeslot` `datetime` on a separate ticket, or as part of this PR?

**Comment by appleJax** [comment]
@gabrielbonner I guess we can deal with it in a separate PR.

**Comment by jeremyjung** [comment]
@appleJax What are you proposing the controller return?  You can add additional calculated attributes to the TimeslotSerializer if you want rails to do some of the lifting for you.

**Comment by gabrielbonner** [comment]
@appleJax Up to you.  This PR looks great to me, if you want to merge it and open a small ticket for the `datetime` formatting, cool, if not, I don't think you're really blocking anyone at the moment so go for it.

**Comment by appleJax** [comment]
@jeremyjung I was thinking something like this:
```
@timeslot = {
  month: "September",
  day_of_month: 12,
  day_of_week: "Tuesday",
  start_time: "1:30PM",
  end_time: "2:30PM"
}
```

**Comment by appleJax** [comment]
I dunno, I guess unless we also include the original time format, it will be harder to sort timeslots on the client. Then again, should we sort them inside the model, that way the client can just display them as is? I'm inclined to just merge this for now, and play around with different configurations in another PR.

**Comment by jeremyjung** [comment]
You can go ahead and merge.  Personally, I think it is ok for the client to do its own parsing of the data but I am ok either way.

If you want rails to take care of it then you can change the TimeslotSerializer to add those individual fields, then when the controller sends the model to the react component, it will have all of those fields.  You don't have to modify the controller.  

If you want an example, look at the `timeslot_serializer.rb` file and look at the computed start_time property.  It serializes a start_time field that is the number of seconds since the epoch which is not a field that originally existed in the model.  You can add attributes and corresponding functions for things like day_of_month, day_of_week, etc..

**Comment by appleJax** [comment]
Ok awesome, I will play around with it...

**Pull request merged**